src/db.rs

```diff
@@ -1286,8 +1298,21 @@ impl Db {
     /// ```
     pub async fn flush(&self) -> Result<(), SlateDBError> {
         if self.inner.wal_enabled() {
-            self.inner.flush_wals().await
```

Comment:
I don't think we need to change this if block at all.

First a quick refresher on the write/flush process.:

When the WAL is enabled, writes go into the wal table. The wal flush task periodically (by default every 100 ms) freezes the wal table and flushes the contents to a new WAL SST. It then writes the contents of the WAL SST into the current memtable (so when WAL is enabled, the memtable always contains data that is durably stored). Once the memtable reaches a size threshold (64MB by default), the wal flush task freezes it and notifies the memtable flush task, which then goes and flushes it.

When the WAL is disabled, writes go directly into the memtable. In this case, the data in the memtable is not durable yet. When a write causes the memtable to cross its size threshold, the writing task freezes the memtable and notifies the memtable flush task which goes and flushes it.

Therefore:
(1) we should not be freezing the memtable here. When wal is enabled, the memtable should only ever be frozen when it reaches its size threshold. There's no reason to flush it here because the data in the memtable is already durably flushed (since the WAL is enabled)
(2) We should just call `self.inner.flush_wals` to flush the WAL. I'm worried `DbInner.flush` isn't safe to call concurrently from multiple tasks. It's assumed to only ever be called by the wal flush task.

Reply:
This PR became what it is now because it originally introduced a helper function called `force_flush` for testing. I was adding tests for #307 and found that calling `flush` did not observe writes in sst. Personally, I would prefer that `DB::flush` clears the contents of memory rather than just persisting the WAL, which is also more in line with the consensus.
![flush](https://github.com/user-attachments/assets/10b9c4ec-1ade-4373-98f2-7143feb8c60c)
[Dissecting, Designing, and Optimizing LSM-based Data Stores](https://cs-people.bu.edu/mathan/publications/sigmod22-sarkar-tutorial.pdf)

Reply:
So if we choose not to flush the contents of the memtables when WAL is enabled, then I would suggest introducing a helper function for testing only as the original pr content. this way we don't need to build test cases for unit tests (if you observe the tests related to flush, many of them need to be artificially constructed to achieve the freeze threshold)

Reply:
> This PR became what it is now because it originally introduced a helper function called force_flush for testing. I was adding tests for https://github.com/slatedb/slatedb/pull/307 and found that calling flush did not observe writes in sst.

Ah I see - so you want to be able to force the memtable to be written so you can validate the contents of the SST? I think for this use case it would make sense to add a different helper function. I don't think it makes sense to roll this into the flush that the user calls. I think the use case of the public flush fn is just to ensure that the data is durably stored - they don't care whether or not some of it resides in memory because the WAL->L0 compaction done by the writer is in progress.

Reply:
> This PR adds a small helper function just for testing. Regardless of whether the current in-memory components(WAL & Memtable) reach the flush threshold, it will force all data to be flushed and persisted.

> Why we need it?
Before introducing this function, many tests such as compaction required carefully constructed test cases to allow the data to reach the flush threshold. This has brought us a lot of inconvenience, this function can reduce the associated mental load.

> slateDB-go has also recently introduced similar functionality.
[slateDB-go #30](https://github.com/slatedb/slatedb-go/pull/30/files)

So I think it is better to follow the original design and introduce a helper function that is only used for testing. This is the original PR content.